Rewatch.lua

```diff
@@ -1403,130 +1424,142 @@ rewatch_events:SetScript("OnEvent", function(timestamp, event, unitGUID, effect,
-			-- update the spellbar
-			rewatch_UpdateBar(spell, targetName, stacks);
-		-- if it's innervate that we cast, report
-		elseif((meGUID == UnitGUID("player")) and (spell == rewatch_loc["innervate"]) and (targetName ~= UnitName("player"))) then
-			SendChatMessage("Innervating "..targetName.."!", "SAY");
-		-- if it's a spell that needs custom highlighting, notify by highlighting player frame	
-		elseif(rewatch_ProcessHighlight(spell, targetName, "Highlighting", "Notify")) then
-		elseif(rewatch_ProcessHighlight(spell, targetName, "Highlighting2", "Notify2")) then
-		elseif(rewatch_ProcessHighlight(spell, targetName, "Highlighting3", "Notify3")) then
-			-- ignore further, already processed it		
-		-- else, if it was a debuff applied
-		elseif(school == "DEBUFF") then
-			-- get the player position, or if -1, return
-			playerId = rewatch_GetPlayer(targetName);
-			if(playerId < 0) then return; end;
-			-- get the debuff type
-			_, _, _, _, debuffType = UnitDebuff(targetName, spell);
-			-- process it
-			if((debuffType == "Curse") or (debuffType == "Poison") or (debuffType == "Magic" and rewatch_loadInt["InRestoSpec"])) then
-				rewatch_bars[playerId]["Corruption"] = spell; rewatch_bars[playerId]["CorruptionType"] = debuffType;
-				if(rewatch_bars[playerId].Buttons[rewatch_loc["removecorruption"]]) then rewatch_bars[playerId].Buttons[rewatch_loc["removecorruption"]]:SetAlpha(1); end;
-				if(rewatch_bars[playerId].Buttons[rewatch_loc["naturescure"]]) then rewatch_bars[playerId].Buttons[rewatch_loc["naturescure"]]:SetAlpha(1); end;
-				rewatch_SetFrameBG(playerId);
-			end;
-		-- else, if it was a bear/cat shapeshift
-		elseif((spell == rewatch_loc["bearForm"]) or (spell == rewatch_loc["direBearForm"]) or (spell == rewatch_loc["catForm"])) then
-			-- get the player position, or if -1, return
-			playerId = rewatch_GetPlayer(targetName);
-			if(playerId < 0) then return; end;
-			-- if it was cat, make it yellow
-			if(spell == rewatch_loc["catForm"]) then
-				val = rewatch_GetPowerBarColor("ENERGY");
-				rewatch_bars[playerId]["ManaBar"]:SetStatusBarColor(val.r, val.g, val.b, 1);
-			-- else, it was bear form, make it red
-			else
-				val = rewatch_GetPowerBarColor("RAGE");
-				rewatch_bars[playerId]["ManaBar"]:SetStatusBarColor(val.r, val.g, val.b, 1);
-			end;
-		-- else, if it was Clearcasting being applied to us
-		elseif((spell == rewatch_loc["clearcasting"]) and (targetName == UnitName("player"))) then
-			for n=1,rewatch_i-1 do val = rewatch_bars[n]; if(val) then
-				if(val[rewatch_loc["regrowth"]]) then
-					val[rewatch_loc["regrowth"].."Bar"]:SetStatusBarColor(1, 1, 1, 1);
+	-- buff applied/refreshed 
+	elseif(event == "COMBAT_LOG_EVENT_UNFILTERED") then
```

Comment:
I'm not really sure what's happening here, the GitHub changelog has some issues with the spacing/lining that has changed. Another patch I've seen has a shorter solution, however, I cannot judge if that captures all your changes below. What are your thoughts?

https://github.com/coenvdwel/Rewatch/blob/61000/Rewatch.lua#L1450

Reply:
I'm glad I'm not the only one who had difficulties following using haha.

Blizzard have made some changes with the COMBAT_LOG_EVENT in that it no longer has any payload. CombatLogGetCurrentEventInfo() is to be used to retrieve that.

The principal difference between my changes and the one mentioned is that, from what I have observed in game, the if statements using the 'effect' variable are checking the combat log and as such are only (at least the ones we are checking for) present when COMBAT_LOG_EVENT is raised, hence we only need to make those checks if the event in question is a COMBAT_LOG_EVENT.

What I have done is wrap those elseif statements in the elseif you have referenced above.

I accept that my implementation when setting the variables might not be as elegant as the other solution, I make no comment on that point and leave that to your best judgement.

To summarise, this change is made to prevent unnecessary checks when an event is raised that is not a COMBAT_LOG_EVENT.

See [8.0 Release Notes for Wow Addon Developers - Blue post](https://us.battle.net/forums/en/wow/topic/20762318007#post-1)